server/src/main/java/org/opensearch/rest/action/search/RestCreatePitAction.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.search;
+
+import org.opensearch.action.search.CreatePitRequest;
+import org.opensearch.action.support.IndicesOptions;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.Strings;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestStatusToXContentListener;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+import static org.opensearch.rest.RestRequest.Method.POST;
+
+/**
+ * Rest action for creating PIT context
+ */
+public class RestCreatePitAction extends BaseRestHandler {
+    public static String ALLOW_PARTIAL_PIT_CREATION = "allow_partial_pit_creation";
+    public static String KEEP_ALIVE = "keep_alive";
+
+    @Override
+    public String getName() {
+        return "create_pit_action";
+    }
+
+    @Override
+    public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        boolean allowPartialPitCreation = request.paramAsBoolean(ALLOW_PARTIAL_PIT_CREATION, true);
+        String[] indices = Strings.splitStringByCommaToArray(request.param("index"));
+        TimeValue keepAlive = request.paramAsTime(KEEP_ALIVE, null);
+        CreatePitRequest createPitRequest = new CreatePitRequest(keepAlive, allowPartialPitCreation, indices);
```

Comment:
Should there be a fault keep alive in case nothing is passed

Reply:
In design doc, keep alive is purposefully made mandatory so that customer is aware of the expiry. Should we change the behavior ?